optuna/study.py

```diff
@@ -472,14 +473,34 @@ def create_study(
         direction:
             Direction of optimization. Set ``minimize`` for minimization and ``maximize`` for
             maximization. Note that ``maximize`` is currently unsupported.
+        exist_ok:
+            Flag for controlling the behavior when study names conflict.
+            If a study named ``study_name`` already exists in the ``storage``,
+            a :class:`~optuna.structs.DuplicatedStudyError` is raised if ``exist_ok`` is
+            set to :obj:`False`.
+            Otherwise the creation of the study is skipped and no exception is raised.
 
     Returns:
         A :class:`~optuna.study.Study` object.
 
     """
 
     storage = storages.get_storage(storage)
-    study_name = storage.get_study_name_from_id(storage.create_new_study_id(study_name))
+    try:
+        study_id = storage.create_new_study_id(study_name)
+    except structs.DuplicatedStudyError:
+        if exist_ok:
+            assert study_name is not None
```

Comment:
Do we need this assertion? I think we can check the scenario in unit tests.

Reply:
If we delete this line and `study_name = None` is passed,  the execution of the code will end with an error raised by https://github.com/pfnet/optuna/blob/0448f096bd6c40c0089f013d90131fad887cd1e3/optuna/storages/rdb/models.py#L52.

Because the deletion of the line does not cause a critical problem (e.g., data inconsistency), I think we can delete it safely.

But I copied this assertion from https://github.com/pfnet/optuna/pull/216/files#diff-4e020643693eb296936f5dd7194bd6d6R397, so I would like to hear @g-votte 's opinion.

Reply:
That makes sense. If a name generated automatically is duplicated, this condition (`study_name=None, exist_ok=True`) will be satisfied. In this case, I don't think `ValueError(NOT_FOUND_MSG)` is an appropriate message.
Thank you for your kind explanation.